src/services/syncFeedPerUser.js

```diff
@@ -152,13 +152,15 @@ const syncMainFeedF2 = async (userId) => {
   console.log('END sync main_feed_f2');
 };
 
-const syncMainFeedBroad = async (userId) => {
+const syncMainFeedBroad = async (userId, source = '') => {
   console.log('START sync main_feed_broad');
   const relatedUserIds = await findRelatedUserIds(userId);
   const unrelatedUserIds = await findUnrelatedUserIds(relatedUserIds);
   const unrelatedAnonUserIds = await findUnrelatedAnonUserIds(relatedUserIds);
   console.log('***** Start Process Feed Board *****');
-  await unFollowMainFeedBroad(userId, relatedUserIds);
+  if (source !== 'registerUser') {
+    await unFollowMainFeedBroad(userId, relatedUserIds);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new optional parameter `source` is introduced to control the execution of `unFollowMainFeedBroad`. However, it's not clear what happens when `source` is not 'registerUser'. If there are other sources that could call this function, they might unintentionally skip the `unFollowMainFeedBroad` operation. It would be better to make the condition more explicit by checking if the source is one of the expected values or by using a boolean flag instead of a string.
```diff
-  if (source !== 'registerUser') {
+  if (source === 'someOtherSource') {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->